mne/time_frequency/tfr.py

```diff
@@ -1374,3 +1376,112 @@ def tfr_multitaper(inst, freqs, n_cycles, time_bandwidth=4.0,
         out = (out, AverageTFR(info, itc, times, freqs, nave,
                                method='mutlitaper-itc'))
     return out
+
+
+def grand_average(all_tfr, drop_bads=True):
+    """Make grand average of a list TFR data
+
+    Selects the subset of good channels across all datasets.
+
+    The grand_average.nave attribute will be equal the number
+    of datasets used to calculate the grand average.
+
+    Note: Grand average TFR shall not be used for source localization.
+
+    Parameters
+    ----------
+    all_tfr : list of AverageTFR data
+        The TFR datasets.
+    drop_bads : bool
+        If True, bad MEG and EEG channels are dropped.
+
+    Returns
+    -------
+    grand_average : AverageTFR
+        The grand average data.
+
+    Notes
+    -----
+    .. versionadded:: 0.10.0
+    """
+    # check if all elements in the given list are evoked data
+    if not all(isinstance(tfr_, AverageTFR) for tfr_ in all_tfr):
+        raise ValueError("Not all the elements in list are AllTFR data")
+
+    # Copy channels to leave the original evoked datasets intact.
+    all_tfr = [tfr_.copy() for tfr_ in all_tfr]
+
+    # Interpolates if necessary
+    if drop_bads:
+        for tfr in all_tfr:
+            tfr.drop_channels(tfr.info['bads'], copy=False)
+    else:
+        pass
+        # implement good channel subset here somehow??
```

Comment:
That's my placeholder for doing channel-wise subject subset averaging ... I think I should raise a NotImplementedError here?